File: frontend/src/services/dashboard.service.ts
// Dashboard Service
// Acts as a proxy layer between components and repositories

import type { IDashboardRepository } from "@/repositories";
import type {
  PaginatedResponse,
  DashboardMetricDTO,
  DashboardOverviewDTO,
  AlertDTO,
  ActivityDTO,
  AlertSearchParams,
  ActivitySearchParams,
} from "@/types";

export class DashboardService {
  constructor(private repository: IDashboardRepository) {}

  /**
   * Get dashboard metrics
   */
  async getMetrics(): Promise<DashboardMetricDTO[]> {
    return this.repository.getMetrics();
  }

  /**
   * Get enterprise dashboard overview
   */
  async getOverview(
    window: "24h" | "7d" | "30d" = "7d"
  ): Promise<DashboardOverviewDTO> {
    return this.repository.getOverview(window);
  }

  /**
   * Get active alerts
   */
  async getAlerts(limit?: number): Promise<AlertDTO[]> {
    return this.repository.getAlerts(limit);
  }

  /**
   * Get recent activities
   */
  async getActivities(limit?: number): Promise<ActivityDTO[]> {
    return this.repository.getActivities(limit);
  }

  /**
   * Acknowledge an alert
   */
  async acknowledgeAlert(id: string): Promise<void> {
    return this.repository.acknowledgeAlert(id);
  }

  /**
   * Resolve an alert
   */
  async resolveAlert(id: string): Promise<void> {
    return this.repository.resolveAlert(id);
  }

  /**
   * Get paginated alerts with filters
   */
  async getAlertsPaginated(
    params: AlertSearchParams
  ): Promise<PaginatedResponse<AlertDTO>> {
    return this.repository.getAlertsPaginated(params);
  }

  /**
   * Get paginated activities with filters
   */
  async getActivitiesPaginated(
    params: ActivitySearchParams
  ): Promise<PaginatedResponse<ActivityDTO>> {
    return this.repository.getActivitiesPaginated(params);
  }
}

File: FhirHubServer/src/FhirHubServer.Core/Services/DashboardService.cs
using FhirHubServer.Core.DTOs.Common;
using FhirHubServer.Core.DTOs.Dashboard;
using FhirHubServer.Core.Interfaces;

namespace FhirHubServer.Core.Services;

public class DashboardService
{
    private readonly IDashboardRepository _repository;

    public DashboardService(IDashboardRepository repository)
    {
        _repository = repository;
    }

    public Task<IEnumerable<DashboardMetricDto>> GetMetricsAsync(CancellationToken ct = default)
        => _repository.GetMetricsAsync(ct);

    public Task<DashboardOverviewDto> GetOverviewAsync(string? window = null, CancellationToken ct = default)
        => _repository.GetOverviewAsync(window, ct);

    public Task<IEnumerable<AlertDto>> GetAlertsAsync(int limit, CancellationToken ct = default)
        => _repository.GetAlertsAsync(limit, ct);

    public Task<IEnumerable<ActivityDto>> GetActivitiesAsync(int limit, CancellationToken ct = default)
        => _repository.GetActivitiesAsync(limit, ct);

    public Task AcknowledgeAlertAsync(string id, CancellationToken ct = default)
        => _repository.AcknowledgeAlertAsync(id, ct);

    public Task ResolveAlertAsync(string id, CancellationToken ct = default)
        => _repository.ResolveAlertAsync(id, ct);

    // Paginated queries
    public Task<PaginatedResponse<AlertDto>> GetAlertsPaginatedAsync(AlertSearchParams searchParams, CancellationToken ct = default)
        => _repository.GetAlertsPaginatedAsync(searchParams, ct);

    public Task<PaginatedResponse<ActivityDto>> GetActivitiesPaginatedAsync(ActivitySearchParams searchParams, CancellationToken ct = default)
        => _repository.GetActivitiesPaginatedAsync(searchParams, ct);
}
